File: 0056-merge-intervals/0056-merge-intervals.java
class Solution {
    public int[][] merge(int[][] intervals) {
        int n = intervals.length;
        Arrays.sort(intervals, (a,b) -> a[0]-b[0]);
        List<int[]> ans = new ArrayList<>();
        int[] temp = intervals[0];

        for(int i = 1; i<n; i++){
            if(intervals[i][0] <= temp[1]){
                temp[1] = Math.max(intervals[i][1],temp[1]);
            }else{
                ans.add(temp);
                temp = intervals[i];
            }
        }
        ans.add(temp);
        return ans.toArray(new int[ans.size()][]);
    }
}

File: 0056-merge-intervals/0056-merge-intervals.cpp
class Solution {
public:
    vector<vector<int>> merge(vector<vector<int>>& intervals) {
        vector<vector<int>> ans;
        int n = intervals.size();
        sort(intervals.begin(), intervals.end());
        for(int i = 0; i<n; i++){
            if( i+1< n && intervals[i][1] >= intervals[i+1][0]){
                int start = min(intervals[i][0], intervals[i+1][0]);
                int end = max(intervals[i][1], intervals[i+1][1]);
                ans.push_back({start, end});
                i++;
            }else{
                ans.push_back(intervals[i]);
            }
        }
        
        return ans;
    }
};